napari/layers/_layer_actions.py

```diff
@@ -239,3 +239,40 @@ def _project(ll: LayerList, axis: int = 0, mode: str = 'max') -> None:
     )
 
     ll.append(new)
+
+
+def _enable_measure_shapes(ll: LayerList):
+    def polygon_area(pts):
+        x, y = pts.T
+        correction = x[-1] * y[0] - y[-1] * x[0]
+        main_area = np.dot(x[:-1], y[1:]) - np.dot(y[:-1], x[1:])
+        return 0.5 * np.abs(main_area + correction)
```

Comment:
Which formula did you use here?

Reply:
this one: https://stackoverflow.com/a/49129646/7836710

I just picked *something* to test the functionality, feel free to suggest something better!

Reply:
If I correctly understand, this is https://en.wikipedia.org/wiki/Shoelace_formula, so it will not fit well to our cases when we do not know orientation and there may be intersections.

I suggest using triangulation output to calculate area.